src/meta.rst

```diff
@@ -71,39 +71,128 @@ A full description of selectors is
 `in the conda docs <http://conda.pydata.org/docs/building/meta-yaml.html#preprocessing-selectors>`_.
 
 
-Using the Compilation ``toolchain``
------------------------------------
-Packages that need to compile code should add the ``toolchain`` package as a build dependency. This helps set up environment variables to instruct the compilers to do the right thing (e.g. building against the right version of the MacOS SDK).
+Pinning packages
+----------------
+
+Conda-smithy 3.0.0 switches to ``conda-build=3``. In conda-smithy 3.0.0, we use a central configuration file from conda-forge-pinning for the build matrices and versions of specific packages.
+
+When a rerendering happens, conda-smithy will render the recipe using conda-build and output configuration files for each job and save them in a yaml file in ``.ci_support`` folder. For example there's a output configuration file for each OS, each python version, etc.
+
+These output configuration files are stripped to options that are used in the build and therefore a change in the config files in ``.ci_support`` folder implies that there needs to be a new build.
+
+Pinning of packages are handled by the same configuration file and conda-build. This means that packages need not be pinned manually. For eg:
+
+.. code-block:: yaml
+
+    requirements:
+      host:
+        - gmp 6.1.*
+      run:
+        - gmp 6.1.*
+
+can be replaced by
+
+.. code-block:: yaml
+
+    requirements:
+      host:
+        - gmp
+
+Note that gmp need not be in ``run`` requirements as ``run_exports`` feature of conda-build will add it automatically.
+
+When there's a new ABI version of gmp (say 7.0), then conda-forge-pinning will be updated. A rerendering of a package using gmp will change. Therefore to check that a recipe needs to be rebuilt for updated pinnings, you only need to check if the package needs a rerender.
+
+Note that ``boost`` and ``numpy`` are exceptions to this. See ``Building Against NumPy`` section.
+
+
+Build matrices
+---------------------
+
+``conda-forge.yml``'s build matrices is removed in conda-smithy=3. To get a build matrix, create a ``conda_build_config.yaml`` file inside recipe folder. For example following will give you 2 builds and you can use the selector ``vtk_with_osmesa`` in the ``meta.yaml``
+
+.. code-block:: yaml
+
+    vtk_with_osmesa:
+      - False
+      - True
+
+You need to rerender the feedstock after this change.
+
+Currently, ``python, vc, r-base`` will create a matrix of jobs for each supported version. If ``python`` is only a build dependency and not a runtime dependency (eg: build script of the package is written in Python, but the package is not dependent on python), use ``build`` section
+
+Following implies that ``python`` is only a build dependency and no Python matrix will be created.
+
+.. code-block:: yaml
+
+    build:
+      - python
+    host:
```

Comment:
@msarahan, from what you said, `host` here is needed right? Otherwise `build == host` and there will be a matrix right?

Reply:
it's this issue: https://github.com/conda/conda-build/issues/2806

Reply:
That's solved in `conda-build` 3.9.2+, right? Does that mean we can drop the empty `host`?